grow/pods/pods.py

```diff
@@ -125,6 +126,17 @@ def _parse_yaml(self):
                 raise PodDoesNotExistError('Pod not found in: {}'.format(path))
             raise podspec.PodSpecParseError('Error parsing: {}'.format(path))
 
+    def set_env(self, env):
+        if env.name:
+            untag = document_fields.DocumentFields.untag
+            content = untag(self.yaml, env_name=env.name)
+            self._yaml = content
```

Comment:
I think this will cause issues if you ever change the environment more than once. The content will become filtered the first time that you `set_env()` and later attempts will always be the same since there are no longer any tagged versions.

Use the `_parse_yaml` (since it is memoized) as the source for all environment untagging instead of `self.yaml`?